toolchain/lc-compile/src/emit.cpp

```diff
@@ -1719,3 +1729,253 @@ void OutputEnd(void)
+        fprintf(stdout, "  is_interface = %d\n", s_depend_mappings[i] . is_interface);
+        fprintf(stdout, "  source_time = %ld\n", s_depend_mappings[i] . source_time);
+        fprintf(stdout, "  interface_time = %ld\n", s_depend_mappings[i] . interface_time);
+    }
+    
+    for(int i = 0; i < s_depend_dep_count; i++)
+    {
+        const char *t_module_name_string;
+        const char *t_dependency_name_string;
+        GetStringOfNameLiteral(s_depend_deps[i] . module, &t_module_name_string);
+        GetStringOfNameLiteral(s_depend_deps[i] . dependency, &t_dependency_name_string);
+        fprintf(stdout, "%s:%s\n", t_module_name_string, t_dependency_name_string);
+    }
+#endif
+    
+    if (DependencyMode < 3)
+    {
+        for(int i = 0; i < s_depend_mapping_count; i++)
+            DependProcess(s_depend_mappings[i] . module);
+    }
+    else if (DependencyMode == 3)
+    {
+        const char *t_output_file;
+        GetOutputFile(&t_output_file);
+        if (t_output_file == NULL)
+        {
+            for(int i = 0; i < s_depend_mapping_count; i++)
+            {
+                DependMapping *t_module;
+                t_module = &s_depend_mappings[i];
+                
+                fprintf(stdout, "%s:", t_module -> interface);
+                for(int j = 0; j < s_depend_dep_count; j++)
+                {
+                    if (s_depend_deps[j] . module != t_module -> module)
+                        continue;
+                    
+                    DependMapping *t_dependency;
+                    if (!DependFindMapping(s_depend_deps[j] . dependency, &t_dependency))
+                        continue;
+                    
+                    fprintf(stdout, " %s", t_dependency -> interface);
+                }
+                
+                if (!t_module -> is_interface)
+                    fprintf(stdout, " %s", t_module -> source);
+                
+                fprintf(stdout, "\n");
+            }
+        }
+        else
+        {
+            // If we have an output code file we just process the first module
+            // and generate an appropriate make-style depedency list.
+            
+            DependMapping *t_module;
+            t_module = &s_depend_mappings[0];
+            
+            fprintf(stdout, "%s: %s\n", t_module -> interface, t_output_file);
+            for(int j = 0; j < s_depend_dep_count; j++)
+            {
+                if (s_depend_deps[j] . module != t_module -> module)
+                    continue;
+                
+                DependMapping *t_dependency;
+                if (!DependFindMapping(s_depend_deps[j] . dependency, &t_dependency))
+                    continue;
+                
+                fprintf(stdout, "%s: %s\n", t_output_file, t_dependency -> interface);
+            }
+        }
+    }
+}
+
+static time_t time_of_file(const char *p_filename)
+{
+    struct stat t_stat;
+    if (stat(p_filename, &t_stat) == -1)
```

Comment:
Err - slow in what context? Given the compiler will have already loaded the entire file and parsed it... I'm not sure the time of a 'stat' call is at all relevant.